src/newbank/server/NewBank.java

```diff
@@ -42,7 +42,62 @@ public static NewBank getBank() {
         return bank;
     }
 
+    //TODO: not needed any more
+    /**
+     * Adds the testing data to the customer HashMap
+     */
+    private void addTestData() throws Exception {
+        Customer bhagy = new Customer("Bhagy", "bhagy");
+        bhagy.addAccount(newAccount("Main", 1000.0));
+        bhagy.addAccount(newAccount("Savings", 1000.0));
+        bhagy.addAccountInfo("100 Test Road, W1 4HJ", "+4471234 663300", "Bhagyashree Patil", "What was your first pet's name?");
+        customers.put(bhagy.getUserName(), bhagy);
+        users.writeUser(bhagy);
+
+
+        Customer christina = new Customer("Christina", "christina");
+        christina.addAccount(newAccount("Savings", 1500.0));
+        customers.put(christina.getUserName(), christina);
+        users.writeUser(christina);
+
+        Customer john = new Customer("John", "john");
+        john.addAccount(newAccount("Checking", 250.0));
+        john.addAccount(newAccount("Savings", 10000));
+        john.getAccount("Checking").addTransaction("Checking", "Test", 100);
+        john.getAccount("Checking").addTransaction("Savings", "Test", 1000);
+        customers.put(john.getUserName(), john);
+        users.writeUser(john);
+        //users.writeMapToFile(customers);
+    }
+    /**
+     * New user sign up
+     */
+    public void newCustomerSignup (String userName, String password, String address, String email) throws Exception{
+        Customer newCustomer = new Customer(userName, password);
+        newCustomer.addAccount(newAccount("Current Account", 0.0));
+        customers.put(newCustomer.getUserName(), newCustomer);
+        users.writeUser(newCustomer);
+    }
+
+    /**
+     *
+     * @param password
+     * @return if the password of the new user satisfies the complexity requirement
+     */
+    public boolean passWordComplexity(String password){
+        return (password.length()>=8);
+    }
+    /**
+     * To hash the user password for database storage.
+     * @param password
+     * @return
+     */
+    private String hashPassword(String password){
```

Comment:
Has this been added to be implemented at a later date?

Reply:
Yes, this is to be implemented by other FR(FR1.12); the declaration leaves room for future implement.